othello/apps/games/models.py

```diff
@@ -0,0 +1,173 @@
+import os
+import uuid
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.postgres.fields import ArrayField
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.db.models import Q
+from django.utils import timezone
+
+from ...moderator.moderator import Player
+
+PLAYER_CHOICES = (
+    (Player.BLACK.value, "Black"),
+    (Player.WHITE.value, "White"),
+)
+
+
+def _save_path(instance, filename):
+    return os.path.join(instance.user.short_name, f"{uuid.uuid4()}.py")
+
+
+class SubmissionSet(models.QuerySet):
+    def latest(self, user=None, **kwargs):
+        if user is not None:
+            qs = self.filter(user=user, **kwargs).order_by("-created_at")
+            return qs[0] if qs.exists() else None
+        else:
+            return self.filter(**kwargs).order_by("user", "-created_at").distinct("user")
```

Comment:
I'm confused. What is this supposed to do?

In fact, what does this method do? It would benefit greatly from a docstring.

Reply:
If `user` is passed, it will return the latest submission for that user.
Else, returns a set of all the latest submissions for all users.

Reply:
This should be split into two methods. The return type of a method should not change based on the value of its parameters, except in very special cases.

Reply:
Removed the user part b/c I don't use it anyways